### PR TITLE
feat: restore legacy API 

### PR DESCRIPTION
**Pull request by sashuk**
[FX-3997]

### Description

This pull request establishes the legacy API so the ReactDayPicker update in Picasso does not introduce any breaking changes and requires no actions from consumers.

#### `DatePicker` / `Calendar`

- TopSchedulerFrontend ([source link](https://github.com/toptal/top-scheduler-frontend/blob/b90f8db9a2f1215152139aae3fdbec8f66976448/src/modules/timeslots/components/DatePicker/DatePicker.tsx#L21))
  - check out https://github.com/toptal/top-scheduler-frontend/pull/435
  - run `yarn storybook`
  - visit http://localhost:6006/?path=/story/timeslots-components-datenavigator--default and http://localhost:6006/?path=/story/timeslots-components-datepicker--default stories

- TopScheduler ([source link](https://github.com/toptal/top-scheduler/blob/b035e46e75c297b6c0938abd5ce3b72219d77ab8/app/front/src/scheduler/components/time_slot_selector/content/table/mobile/__specs__/index_spec.jsx#L99)) Not related to Picasso DatePicker

Props to check (`renderRoot()`, `renderMonthHeader()`, `renderDay()`)

#### `renderDay()`

- TopScheduler ([source link](https://github.com/toptal/top-scheduler/blob/b035e46e75c297b6c0938abd5ce3b72219d77ab8/app/front/src/scheduler/components/time_slot_selector/content/table/mobile/__specs__/index_spec.jsx#L99)) Not related to Picasso DatePicker

- ClientPortal ([source link for scheduling a break](https://github.com/toptal/client-portal/blob/e7e49b6d032647e701d7195d2fd593ca2adf910f/hosts/client-portal/containers/picassoModals/ScheduleEngagementBreak/Content/index.tsx#L133) and [source link for ending engagement](
https://github.com/toptal/client-portal/blob/a2a43bebb6e8afe0f01b2d7b510d062421050ff5/hosts/client-portal/containers/picassoModals/OffboardingWizardModal/components/EndEngagement/components/EndEngagementContent/EndEngagementForm/index.tsx#L265))
  - check out https://github.com/toptal/client-portal/pull/6630 or visit https://client-portal-pr-6630.toptal.rocks/jobs/VjEtSm9iLTMxNTYwNw
  - run `yarn start:staging`, login as `Bruen, Deckow and Tremblay` client
  - schedule a break at https://local-development.staging.toptal.net:4004/jobs/VjEtSm9iLTMxNTYwNw, play with Engagement options, compare with `staging` Client Portal
  - end engagement (follow instructions from https://github.com/toptal/client-portal/pull/5389) at https://client-portal-pr-6630.toptal.rocks/engagements/VjEtRW5nYWdlbWVudC0zMzY4NDM

- Client Portal ([source link](https://github.com/toptal/client-portal/blob/a2a43bebb6e8afe0f01b2d7b510d062421050ff5/hosts/client-portal/containers/pages/Lobby/CallbackForm/index.tsx#L255)) Is not related to Picasso DatePicker

- Platform ([source link](https://github.com/toptal/platform/blob/7baa60b07274f34bc2f41dd581d05ee9f99c4568/app/assets/javascripts/platform/bundles/ui/react_components/calendar/calendar_month.jsx#L164)) Is not related to Picasso DatePicker

#### `renderMonthHeader()` and `renderRoot`

- Staff Portal ([source link](https://github.com/toptal/staff-portal/blob/47241c1aa564ada9213e1fbc7d93da3e04198c99/libs/ui/src/containers/DayOffCalendar/DayOffCalendar.tsx#L42))
  - visit https://staff-portal-pr-10443.toptal.rocks/roles/100010/dayoff (https://github.com/toptal/staff-portal/pull/10443) and compare it with `staging` Staff Portal

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [N/A] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [N/A] Covered with tests

**Breaking change**

- [N/A] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3977]: https://toptal-core.atlassian.net/browse/FX-3977?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FX-3997]: https://toptal-core.atlassian.net/browse/FX-3997?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

File: packages/picasso/src/CalendarContainer/CalendarContainer.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ export type CalendarContainerProps = {
   hasFooter?: boolean
 }
 
-export type RenderRoot = (args: CalendarContainerProps) => ReactNode
+export type RenderRoot = (args: CalendarContainerProps) => JSX.Element
```
Comment (sashuk) [inline packages/picasso/src/CalendarContainer/CalendarContainer.tsx:19]: Keeping type the same as in legacy

File: .changeset/stupid-bears-turn.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+'@toptal/picasso': major
```
Comment (sashuk) [inline .changeset/stupid-bears-turn.md:2]: According to the https://toptal-core.atlassian.net/wiki/spaces/FE/pages/2774728938/Handling+breaking+change, the breaking change is fixed either by updating test snapshots or changing the type of `weekStartsOn` from `number` to `WeekStart` (which is `0 | 1 | ... | 6`), so no need to have temporary deprecation or codemod

File: packages/picasso/src/CalendarDay/CalendarDay.tsx [hunk 3]
```diff
@@ -58,19 +46,22 @@ const CalendarDay = (dayProps: CalendarDayProps): JSX.Element => {
...
+    disabled: isDisabled = false,
+    indicated: isIndicated = false,
+    outside: isOutside = false,
+    weekend: isWeekend = false,
+    temporaryRangeMiddle: isTemporaryRangeMiddle = false,
+    temporaryRangeEnd: isTemporaryRangeEnd = false,
+    range_start: isRangeStart = false,
+    range_middle: isRangeMiddle = false,
+    range_end: isRangeEnd = false,
   } = activeModifiers
```
Comment (sashuk) [inline packages/picasso/src/CalendarDay/CalendarDay.tsx:61]: Almost all modifiers provided by `react-day-picker` are easily convertible to the legacy flags that are later passed to `renderDay()`

File: packages/picasso/src/CalendarMonthHeader/CalendarMonthHeader.tsx [hunk 4]
```diff
@@ -17,7 +17,9 @@ interface RenderMonthHeaderProps extends CaptionProps {
   children?: ReactNode
 }
 
-export type RenderMonthHeader = (args: RenderMonthHeaderProps) => ReactNode
+export type RenderMonthHeader = (
```
Comment (sashuk) [inline packages/picasso/src/CalendarMonthHeader/CalendarMonthHeader.tsx:20]: Keeping type the same as in legacy

File: packages/picasso/src/Calendar/index.ts [hunk 5]
```diff
@@ -1,10 +1,9 @@
 export { default } from './Calendar'
 export * from './Calendar'
-export type { RenderDayProps } from '../CalendarDay'
-export type { DayProps } from 'react-day-picker'
 export type {
   DateOrDateRangeType,
   DateRangeType,
   CalendarDateRange,
   WeekStart,
+  DayProps,
```
Comment (sashuk) [inline packages/picasso/src/Calendar/index.ts:8]: The `DayProps` stays the same as in the previous version of `Calendar` that used `simple-react-calendar`

**Review by TomasSlama — APPROVED** [review]
Looks good!, tested in TopScheduler and in SP

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Review by mkrl — APPROVED** [review]
Awesome job! Looks good

File: packages/picasso/src/CalendarDay/CalendarDay.tsx [hunk 6]
```diff
@@ -80,7 +72,7 @@ const CalendarDay = (dayProps: CalendarDayProps): JSX.Element => {
       data-testid={`day-button-${
         isSelected ? 'selected' : getDayFormatted(date)
       }`}
-      key={key}
```
[the remark this answers is not in the document]
Comment (sashuk) [reply packages/picasso/src/CalendarDay/CalendarDay.tsx:83]: @mkrl Yep, it was removed as `key` is provided by `react-day-picker` on every layer, in this case it is provided for `<td/>` in `node_modules/react-day-picker/src/components/Row/Row.tsx`

```
export function Row(props: RowProps): JSX.Element {
  ...
  return (
      ...
      {props.dates.map((date) => (
        <td
          ...
          key={getUnixTime(date)}
        >
          ...
        </td>
      ))}
  );
}
```

**Pull request merged**